docs/functional/architecture/0011-Scalar_vs_0d_Fields.md

```diff
@@ -0,0 +1,139 @@
+---
+tags: [frontend]
+---
+
+# Scalar vs 0d-fields
+
+- **Status**: valid
+- **Authors**: Hannes Vogt (@havogt)
+- **Created**: 2022-10-03
+- **Updated**: 2022-10-03
+
+Do we distinguish scalars (e.g. `float`) and fields with 0 dimensions (e.g. `Field[[], float]`) in the context of field view?
+
+## Context
+
+In the context of multi-dimensional fields, the question arises if we need scalars **and** 0-d fields, only one of them or even make them the same thing. Similar discussions happened within the Numpy community, see e.g. https://mail.python.org/pipermail/numpy-discussion/2006-February/006384.html.
+
+## Decision
+
+We decided to have both scalars and 0d-fields. They are not treated as alias, but have their distinct properties. Python typing semantics makes it difficult to have a nice solution and we want to comply with Python semantics for typing, especially for embedded execution.
+
+### Examples
+
+#### Wrong annotation for return type
+
+The following is forbidden because the return type annotation doesn't match the type of `1.0` (the correct annotation would be `float`).
+Note: the annotation (by Python design) is for the function that is being decorated, not for the function after the decorator is applied, see also this discussion https://github.com/python/typing/issues/412.
+
+```python=
+@field_operator
+def foo() -> Field[[], float]
+    return 1.0
+```
+
+An explicit broadcast would be required to make this correct.
+
+```python=
+@field_operator
+def foo() -> Field[[], float]
+    return broadcast(1.0, ())
+```
+
+#### Implicit broadcast to Field
+
+In the following call an implicit broadcast to Field is performed in the decorator (the annotation influences behavior).
+
```

Comment:
to make this more explicit (and consistent with the rest) you could add the return type explicitly here
`@field_operator
def foo(scalar: Field[[], float])->Field[[], float]:
    return scalar
`